identity/aziot-identityd/Cargo.toml

```diff
@@ -15,15 +15,18 @@ chrono = "0.4"
 futures-util = "0.3"
 http = "0.2"
 hyper = "0.14"
+futures = "0.3"
```

Comment:
Use only the individual `futures-*` crates that you need instead of the whole `futures` crate. You likely only need `futures-util` for the `{Try,}{Future,Stream}Ext` traits and `futures-core` for `Stream`.

Reply:
Some food for thought: we already take a dependency on the entire `futures` crate as part of `tokio-openssl` (See our `Cargo.toml` [here](https://github.com/Azure/iot-identity-service/blob/1452c9b2d/Cargo.lock#L2100-L2110)). Moreover, given we have so many async ecosystem crates in our dependency graph, I'd be _very_ surprised if we aren't already implicitly bringing in all of the `futures-*` crates via transitive dependencies already.

If the goal is to avoid bringing in unused dependencies, this change won't really change much. That said, it is definately better to be precise precise about what subset of `futures` functionality we are using, so this fix does have merit.

Reply:
Yes, I noticed that about `tokio-openssl` when doing the tokioup. It's doing it wrong. I'm planning to send it a PR when I have the time.

Reply:
Removed